src/qumin/representations/frequencies.py

```diff
@@ -0,0 +1,398 @@
+        elif real and (cls.source['forms'] != "empty"):
+            log.info(f'{name}: No frequencies in the {name} table, building from the forms.')
+            freq = cls.forms.groupby(name[:-1]).value.sum()
+            table.loc[freq.index, "value"] = freq.values
+            table['source'] = 'forms_table'
+            cls.source[name] = 'forms_table'
+
+        # 3. Building a fake uniform frequency distribution.
+        else:
+            log.info(f'{name}: Building empty frequencies.')
+            table['source'] = 'empty'
+            table['value'] = pd.NA
+            cls.source[name] = 'empty'
+
+        # We save the resulting table
+        table.sort_index(inplace=True)
+        table.index.name = name[:-1]
+        setattr(cls, name, table[['value', 'source']])
+
+    @classmethod
+    def get_absolute_freq(cls, mean=False, group_on=False, skipna=False, **kwargs):
+        """
+        Return the frequency of an item for a given source
+
+        The frequency of an item is defined as the sum of the frequencies of this item
+        across all rows.
+
+        Examples:
+
+            >>> p = fl.Package('tests/data/TestPackage/test.package.json')
+            >>> Frequencies.initialize(p, real=True)
+            >>> Frequencies.get_absolute_freq(filters={'lexeme':'q'}, group_on="index", skipna=True)
+            form
+            11    12.0
+            12     6.0
+            14    20.0
+            18     NaN
+            Name: value, dtype: float64
+            >>> Frequencies.get_absolute_freq(filters={'lexeme':'q'})
+            nan
+            >>> Frequencies.get_absolute_freq(filters={'cell':'third'}, mean=True, skipna=True)
+            20.0
+            >>> Frequencies.get_absolute_freq(group_on=['lexeme'])
+            lexeme
+            k    193.0
+            p      NaN
+            q      NaN
+            s     33.0
+            Name: value, dtype: float64
+
+        Arguments:
+            group_on (List[str]): columns on for which absolute frequencies should be computed.
+                If `False, aggregates across all records.
+            mean (bool): Defaults to False. If True, returns a mean instead of a sum.
+            skipna(bool): Defaults to False. Skip nan values for sums or means.
+        Returns:
+            `pandas.Series`: a Series which contains the output values.
+                The index is either the original one, or the grouping columns.
+        """
+
+        # Filter using keys from mapping dict
+        sublist = cls._filter_frequencies(**kwargs)
+
+        if group_on == "index":
+            return sublist.value
+        elif group_on is False:
+            groups = [True] * len(sublist)
+        else:
+            groups = group_on
+
+        if mean:
+            def func(x): return x.mean(skipna=skipna)
+        else:
+            def func(x): return x.sum(skipna=skipna)
+
+        result = sublist.groupby(by=groups, group_keys=False).value.apply(func)
```

Comment:
Wouldn't something like this be cleaner ?

~~~
func_name = "mean" if mean else "sum"
sublist.groupby(by=groups, group_keys=False).agg({"value": func_name}, skipna=skipna)`
~~~

You could even replace "mean=False" in the arg list by `aggfunc="sum"`, that way to get a mean the user can just pass `aggfunc="mean"` (or something else). Up to you.

Reply:
Do check the returned types, `.value.apply` and `.agg({"value":func})` might not return the exact same thing

Reply:
Ugh, ok no problems then to keep as is. I assumed the interface would be the same !